### PR TITLE
Support Route Map CRUD

### PR DESCRIPTION
**Pull request by zaynchen115**
## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    Design PR https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1309
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Network/Network/Cortex/RouteMap/GetAzureRmRouteMapCommand.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Alias("VirtualHubName", "ParentVirtualHubName", "ParentResourceName")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubName,
+            HelpMessage = "The parent resource name.")]
+        [ResourceNameCompleter("Microsoft.Network/virtualHubs", "ResourceGroupName")]
+        public string HubName { get; set; }
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/RouteMap/GetAzureRmRouteMapCommand.cs:44]: VirtualHubName

File: src/Network/Network/Cortex/RouteMap/GetAzureRmRouteMapCommand.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+        [Alias("ParentObject", "ParentVirtualHub")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubObject,
+            HelpMessage = "The parent resource.")]
+        public PSVirtualHub VirtualHub { get; set; }
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/RouteMap/GetAzureRmRouteMapCommand.cs:52]: VirtualHubObject

File: src/Network/Network/Cortex/RouteMap/GetAzureRmRouteMapCommand.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+        [Alias("VirtualHubId", "ParentVirtualHubId")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubResourceId,
+            HelpMessage = "The parent resource id.")]
+        [ResourceIdCompleter("Microsoft.Network/virtualHubs")]
+        public string ParentResourceId { get; set; }
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/RouteMap/GetAzureRmRouteMapCommand.cs:61]: VirtualHubResourceId

File: src/Network/Network/Cortex/RouteMap/NewAzureRmRouteMapCommand.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Alias("VirtualHubName", "ParentVirtualHubName")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubName,
+            HelpMessage = "The resource group name.")]
+        public string ParentResourceName { get; set; }
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/RouteMap/NewAzureRmRouteMapCommand.cs:44]: VirtualHubName

File: src/Network/Network/Cortex/RouteMap/NewAzureRmRouteMapCommand.cs [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+        [Alias("VirtualHub", "ParentVirtualHub")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubObject,
+            HelpMessage = "The parent resource.")]
+        public PSVirtualHub ParentObject { get; set; }
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/RouteMap/NewAzureRmRouteMapCommand.cs:52]: VirtualHubObject

File: src/Network/Network/Cortex/RouteMap/NewAzureRmRouteMapCommand.cs [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+        [Alias("VirtualHubId", "ParentVirtualHubId")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubResourceId,
+            HelpMessage = "The parent resource.")]
+        [ResourceIdCompleter("Microsoft.Network/virtualHubs")]
+        public string ParentResourceId { get; set; }
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/RouteMap/NewAzureRmRouteMapCommand.cs:61]: VirtualHubResourceId

File: src/Network/Network/Cortex/RouteMap/NewAzureRmRouteMapCommand.cs [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+        [Alias("ResourceName", "RouteMapName")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Name of the route map resource.")]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The list of route map rules for this route map resource.")]
+        public PSRouteMapRule[] RouteMapRules { get; set; }
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/RouteMap/NewAzureRmRouteMapCommand.cs:72]: RouteMapRule

File: src/Network/Network/Cortex/RouteMap/RemoveAzureRmRouteMapCommand.cs [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Alias("VirtualHubName", "ParentVirtualHubName")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = CortexParameterSetNames.ByRouteMapName,
+            HelpMessage = "The resource group name.")]
+        public string ParentResourceName { get; set; }
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/RouteMap/RemoveAzureRmRouteMapCommand.cs:45]: VirtualHubName

File: src/Network/Network/Cortex/RouteMap/RemoveAzureRmRouteMapCommand.cs [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubObject,
+            HelpMessage = "Name of the route map resource.")]
+        public string Name { get; set; }
+
+        [Alias("VirtualHub", "ParentVirtualHub")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubObject,
+            HelpMessage = "The parent virtual hub object.")]
+        public PSVirtualHub ParentObject { get; set; }
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/RouteMap/RemoveAzureRmRouteMapCommand.cs:63]: VirtualHubObject

File: src/Network/Network/Cortex/RouteMap/SetAzureRmRouteMapCommand.cs [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Alias("VirtualHubName", "ParentVirtualHubName")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = CortexParameterSetNames.ByRouteMapName,
+            HelpMessage = "The resource group name.")]
+        public string ParentResourceName { get; set; }
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/RouteMap/SetAzureRmRouteMapCommand.cs:44]: VirtualHubName

File: src/Network/Network/Cortex/RouteMap/SetAzureRmRouteMapCommand.cs [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubObject,
+            HelpMessage = "Name of the route table.")]
+        public string Name { get; set; }
+
+        [Alias("VirtualHub", "ParentVirtualHub")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubObject,
+            HelpMessage = "The parent virtual hub object.")]
+        public PSVirtualHub ParentObject { get; set; }
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/RouteMap/SetAzureRmRouteMapCommand.cs:62]: VirtualHubObject

File: src/Network/Network/Cortex/RouteMap/SetAzureRmRouteMapCommand.cs [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CortexParameterSetNames.ByRouteMapResourceId,
+            HelpMessage = "The resource id of the route map to modify.")]
+        [ResourceIdCompleter("Microsoft.Network/virtualHubs/routeMaps")]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The list of route map rules for this route map resource.")]
+        public PSRouteMapRule[] RouteMapRules { get; set; }
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/RouteMap/SetAzureRmRouteMapCommand.cs:84]: RouteMapRule

File: src/Network/Network/Cortex/RouteMap/SetAzureRmRouteMapCommand.cs [hunk 13]
```diff
@@ -0,0 +1,158 @@
+﻿// ----------------------------------------------------------------------------------
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/RouteMap/SetAzureRmRouteMapCommand.cs:1]: please use Update instead of Set

**Review by VeryEarly — CHANGES_REQUESTED** [review]
Add/Get/Set/Remove RouteMapRule do not look necessary. please remove them.

**Review by VeryEarly — CHANGES_REQUESTED** [review]
please generate help markdown and generate test cases for new cmdlets

**Comment by zaynchen115** [comment]
Still working on generate test files, will add help files

**Comment by VeryEarly** [comment]
> Regarding 'Add/Get/Set/Remove RouteMapRule do not look necessary. please remove them.'
> 
> Do you suggest only use New/Update-RouteMap to modify route map rules under it?

I suggest only keep `New` cmdlet since it's only in memory object without any API call.

**Comment by zaynchen115** [comment]
@VeryEarly Could you help review, I don't know why the task timed out.

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**